public/components/trace_analytics/components/common/plots/service_map.tsx

```diff
@@ -149,6 +138,19 @@ export function ServiceMap({
     },
   ];
 
+  // For the traces custom page
+  useEffect(() => {
+    if (!selectableValue || selectableValue.length === 0) {
+      // Set to the first option ("latency") and trigger the onChange function
+      const defaultValue = metricOptions[0].value;
+      setSelectableValue(defaultValue); // Update the state
+      setIdSelected(defaultValue); // Propagate the default to parent
```

Comment:
what's the purpose of this useEffect? it's setting state for idSelected and selectableValue, but it also depends on them. When these states are updated, wouldn't it trigger useEffect again?